pkg/prometheus/agent/operator.go

```diff
@@ -82,14 +83,64 @@ type Operator struct {
 	endpointSliceSupported bool
 }
 
+func Runnable(ctx context.Context, conf operator.Config, logger log.Logger) (bool, error) {
+	cfg, err := createK8sClusterConfig(conf)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating cluster config failed")
+	}
+
+	kclient, err := createK8sClient(cfg)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating kubernetes client failed")
+	}
+
+	crdInstalled, err := k8sutil.IsAPIGroupVersionResourceSupported(kclient.Discovery(), monitoringv1alpha1.SchemeGroupVersion.String(), monitoringv1alpha1.PrometheusAgentName)
+	if err != nil {
+		level.Warn(logger).Log("msg", "failed to check if the API supports the PrometheusAgent CRD", "err ", err)
+		return false, err
+	}
+	if !crdInstalled {
+		level.Info(logger).Log("msg", "Prometheus agent controller disabled because the PrometheusAgent CRD isn't installed")
```

Comment:
In my opinion this should be `Warn` as well 
```suggestion
		level.Warn(logger).Log("msg", "Prometheus agent controller disabled because the PrometheusAgent CRD isn't installed")
```

Reply:
I'm not sure: you might not want to deploy a v1alpha1 CRD on your cluster. In which case, the warning log might might be confusing.
But if the CRD is deployed, not having the required permissions is worth a warning.

Reply:
Yes, that makes sense! It's indeed two separate situations that should be handled differently! Thanks for clarifying!